src/bun.js/bindings/bindings.zig

```diff
@@ -2762,6 +2867,22 @@ pub const JSValue = enum(JSValueReprInt) {
         });
     }
 
+    pub fn asAnyPromise(
+        value: JSValue,
+    ) ?*AnyPromise {
```

Comment:
This shouldn't be a promise, just the value type

Reply:
shouldn't be a pointer

Reply:
The return type should be `?AnyPromise`

Reply:
If I do that, then we get the following compiler error:
```
zig/lib/std/meta.zig:181:17: error: Expected pointer, optional, array or vector type, found 'src.bun.js.bindings.bindings.AnyPromise'
        else => @compileError("Expected pointer, optional, array or vector type, found '" ++ @typeName(T) ++ "'"),
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/bun.js/event_loop.zig:534:59: note: called from here
        return waitForPromiseWithType(this, std.meta.Child(@TypeOf(promise)), promise);
                                            ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
```

Reply:
Change  that line to

```zig
       if (comptime @TypeOf(promise) == JSC.AnyPromise) 
          return waitForPromiseWithType(this, JSC.AnyPromise, promise);

        return waitForPromiseWithType(this, std.meta.Child(@TypeOf(promise)), promise);
```

Reply:
Found a way around this - https://github.com/oven-sh/bun/pull/1697/commits/0323676611ecdefe46d6e334d52dfc7360b2856e